templates/vhost/_file_header.erb

```diff
@@ -3,6 +3,14 @@
 # Managed by Puppet
 # ************************************
 <%= [@comment].flatten.collect{|c| "# #{c}"}.join("\n") -%>
+<% if @mdomain -%>
+
+  <%- if @mdomain.is_a?(String) -%>
+MDomain <%= @mdomain %>
+  <%- else -%>
+MDomain <%= @servername %>
+  <%- end -%>
+<% end -%>
```

Comment:
do you know how well this template will fare with `MDRequireHttps permanent` given that our standard vhost spreads a http and https over two files?

Reply:
I have not yet used `MDRequireHttps`, I rely on the code I was previously using in my `profile::apache` which supposedly does the same.  I guess that I would use `MDRequireHttps temporary` when attempting to change to ensure no side effect occurs…

Are you in a position you can check?

Reply:
On my infra: after removing the shared HTTP VirtualHost that handled redirects to https, setting `MDRequireHttps` and adding a `apache::listen { '80': }` so that the server listen on port 80; the redirect do not happen.

A quick test shows that you _must_ have a HTTP VirtualHost beside each HTTPS VirtualHost like you have, whatever the content…  For now, I reverted to my custom single point of redirection from my apache profile.

Reply:
> Are you in a position you can check?

i'm in a bathtub.

what if you create an HTTP vhost, with the `MDomain` stanza, and without SSL (obviously?).
does the redirect work then?

Reply:
It does redirect to https.  If you have `MDRequireHttps` and a VirtualHost on port 80, it will redirect to https.  No need for a VirtualHost on port 443 :neutral_face:.

Reply:
i think we might need to warn users, either in the code or in the docs about this behaviour

Reply:
My feeling is that it's really a misconfiguration from the user: it really makes no sense to setup a http-only VirutalHost and set MDomain for it…  So a broken server in such a case is clearly what I would expect.  Just like if I set docroot to a directory that does not exist.

Maybe I do not understand your point because during all the tests i did in this conversation I was wondering what you actually wanted to know / where you wanted to go.

Reply:
> it really makes no sense to setup a http-only VirutalHost and set MDomain for it… So a broken server in such a case is clearly what I would expect. Just like if I set docroot to a directory that does not exist.

sorry, that's not what i meant:
I meant that when you setup `MDRequireHttps $redirect`; then you *also* need to setup an http Vhost (with `MDomain`?), not just an https vhost.

> Maybe I do not understand your point because during all the tests i did in this conversation I was wondering what you actually wanted to know / where you wanted to go.

so i  meant that we should warn people when that's not the case, but i now realize that's really hard to do; so the least we can do is put it into the documentation.

Reply:
Ah!  Got it!

@igalic, what about adding a line in the in-line documentation like:

```diff
 #
 # @param md_require_https
 #   Redirects http: traffic to https: for Managed Domains.
+#   An http: Virtual Host must nevertheless be setup for that domain.
 #
 # @param md_server_status
 #   Control if Managed Domain information is added to server-status.
 
```

Reply:
Fixed, thanks!